src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -579,5 +579,37 @@ public static string GetNativeLibsRootDirectoryPath (string androidBinUtilsDirec
 			string relPath = GetToolsRootDirectoryRelativePath (androidBinUtilsDirectory);
 			return Path.GetFullPath (Path.Combine (androidBinUtilsDirectory, relPath, "lib"));
 		}
+
+		public static string? GetAssemblyCulture (ITaskItem assembly)
+		{
+			// The best option
+			string? culture = assembly.GetMetadata ("Culture");
+			if (!String.IsNullOrEmpty (culture)) {
+				return TrimSlashes (culture);
+			}
+
+			// ...slightly worse
+			culture = assembly.GetMetadata ("RelativePath");
```

Comment:
So I checked the history and there was some discussion around RelativePath here: https://github.com/xamarin/xamarin-android/pull/7823#pullrequestreview-1310314464

Not sure how that would've been set for satellite assemblies though, I can't find where it'd be set.

I'm only seeing Culture (for satellite assemblies in the current project) or DestinationSubDirectory (for satellites from referenced projects) being set.

Instead of checking every assembly can you only look at the satellite assemblies? you're already passing those to the GeneratePackageManagerJava task

Reply:
Yeah, we possibly should do that, but not until after https://github.com/xamarin/xamarin-android/pull/8478 is merged (it changes a lot in this area, and all the optimizations and tweaks to the code should be done only after the merge)